### PR TITLE
PDA-325 Create common GHA for PDA services

### PR DESCRIPTION
**Pull request by dcbickfo**
Change List:
- Breaking change on name of eks action to accommodate a second one
- Creation of eks cache composite action; simply wraps github's action/cache@v3  - Why? We are using this in a few places so keeping track of the default path we're using everywhere is less than ideal

### CODE REVIEW

**Comment by dorner** [comment]
Looks good to me - are any other repos using these actions? I.e. is this breaking change only affecting the other actions in this repo or do other repos have to change?

**Comment by dcbickfo** [comment]
There are other PDA repos using this for sure. It'll be a breaking change when the repos update to `v0.11` or whatever we tag this one. I can go through these PDA repos and make the change as part of my current story.

I intend for there to be more changes to this branch before I merge, but wanted an early buy in that the strategy for our loaders (which is the easier set of actions) was looking okay. 😄

**Comment by dorner** [comment]
As long as it works and speeds things up I'm good with it :grin:

**Comment by dcbickfo** [comment]
@dorner - I've gone through each of our go repos with these actions now. I had to make a couple more re-organization pieces based on the difference between our lambda and eks deploys. The go ones are in a good state - I think we can change the `@PDA` tag I'm using throughout to a `v0` tag.

I suggest this instead of `@main` so that we have a clear way to introduce breaking changes (assuming other teams start using these actions). To stick with trunk based development with tags, we can use a strategy like Github is using in `github-actions/checkout` and just update the `v0` tag each time we push a minor release. When we introduce a breaking change it can get tagged as a `v1`. Unfortunately it doesn't look like github-actions respects semantic versioning right now and just pulls the tag.

File: .github/workflows/main-version-update.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Update Main Version
```
Comment (dcbickfo) [inline .github/workflows/main-version-update.yml:1]: Heavily inspired by a github script used in `github-actions/checkout` - the repo is MIT licensed.

**Review by dorner — APPROVED** [review]
Everything looks good to me!

**Comment by dcbickfo** [comment]
Thank you!

I temporarily tagged the latest commit on this branch as `v0` so that I can go through and update all the PDA services to use the latest commit. I'll tag the version as `v0.11` when we merge to main and then update the `v0` tag to point main.

File: go/build/action.yml [hunk 2]
```diff
@@ -1,32 +0,0 @@
-name: 'go-build'
```
Comment (dcbickfo) [inline go/build/action.yml:1]: No workflows use this currently

File: go/configure/action.yml [hunk 3]
```diff
@@ -1,10 +1,7 @@
-name: 'go-setup'
+name: 'go-configure'
```
Comment (dcbickfo) [inline go/configure/action.yml:1]: Renamed to reduce confusion with the old action since this no longer installs go.

**Review by dorner — APPROVED** [review]

**Pull request merged**